### PR TITLE
Add contribution instructions to search-collector README.

### PR DESCRIPTION
**Pull request by mqhackett**
**Related Issue:**  open-cluster-management/backlog#12714

### Description of changes
Add instructions on new PROW based contribution flow. Fork/merge instead of branch/merge

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -25,6 +25,7 @@ import (
 	tr "github.com/open-cluster-management/search-collector/pkg/transforms"
 
 	"github.com/golang/glog"
+	"github.com/open-cluster-management/addon-framework/pkg/lease"
```
Comment (jlpadilla) [inline main.go:28]: Should this be in this PR?

Comment (mqhackett) [reply main.go:28]: This PR is where I'm working on adding the lease. As I began working on the code, I had to setup the fork/merge workflow so I decided to squeeze that documentation into the same PR. This was actually the main thrust of this PR

Comment (jlpadilla) [reply main.go:28]: It sounds different from the PR's title!
In this case it's better to divide it so we can merge the README update and continue with the other work independently.

Comment (mqhackett) [reply main.go:28]: Agreed. Let's use this PR that sherin has created, and keep this current PR small with only the doc updates. 

(I said to myself "I'll change the title to reflect the work once the code reflects the title", but now I'm realizing this PR can be more useful now if merged with doc updates only. No need to wait :) )

Let me remove the package import line.

**Comment by mqhackett** [comment]
Let's work from this PR for adding lease/status work. https://github.com/open-cluster-management/search-collector/pull/132

File: readme.md [hunk 2]
```diff
@@ -49,4 +49,19 @@ RUNTIME_MODE    | no       | production             | Running mode (development
 
 ### Other Configuration Options
 - Environment variables can also be set in ./config.json for development. If both provide a value for a specific property, the environment variable will override the file. You can define your own config json file and pass it to the application with `-c <config_file>`
-- The application can take any flags for [glog](https://github.com/golang/glog), it will pass them straight into glog. The glog flag `--logtostderr` is set to true by default.
\ No newline at end of file
+- The application can take any flags for [glog](https://github.com/golang/glog), it will pass them straight into glog. The glog flag `--logtostderr` is set to true by default.
+
+
+### Contribution
+When making contributions to the search collector project, fork the project then merge your chages by creating a pull request from your personal fork to the main project.
```
Comment (jlpadilla) [inline readme.md:56]: 1. Keep generic to make it easy to copy/paste and maintain across our repos.
2. Fix typo.

```suggestion
When making contributions to this project, fork the project then merge your changes by creating a pull request from your personal fork to the main project.
```

File: readme.md [hunk 3]
```diff
@@ -49,4 +49,19 @@ RUNTIME_MODE    | no       | production             | Running mode (development
...
+
+1. Fork Repository
+2. Clone your fork
+3. Set upstream origin `set upstream origin git@github.com:open-cluster-management/search-collector.git`
+4. Make your edits. Please test them locally before creating a pull request to ensure smooth merging :smile:
```
Comment (jlpadilla) [inline readme.md:61]: Add a pointer to how to test locally.
```suggestion
4. Make your edits. Test locally with `make test` before creating a pull request to ensure smooth merging :smile:
```

**Review by jlpadilla — COMMENTED** [review]
Changes look good for content, but adding @dockerymick to review writing.
/lgtm
/hold

**Comment by dockerymick** [comment]
Thanks for pulling me in @jlpadilla I am taking a look now

File: readme.md (hunk 2 above)
Comment (dockerymick) [reply readme.md:56]: I suggest that the statement reads: 

When you make contributions to this project, fork the project  then merge your changes by creating a pull request from your personal fork to the main project:

File: readme.md (hunk 3 above)
Comment (dockerymick) [reply readme.md:61]: Be sure to have punctuation to end the statements. 

1. Fork the `search-collector` repository.
2. Clone your fork. 
3. Set upstream origin with the following command:

File: readme.md [hunk 4]
```diff
@@ -49,4 +49,19 @@ RUNTIME_MODE    | no       | production             | Running mode (development
...
+5. Push your new commits to your personal fork `git push origin`
```
Comment (dockerymick) [inline readme.md:62]: Push your new commits to your personal fork with the following command:

File: readme.md [hunk 5]
```diff
@@ -49,4 +49,19 @@ RUNTIME_MODE    | no       | production             | Running mode (development
...
+6. Create a pull request from your personal fork again the upstream search-collector main branch
```
Comment (dockerymick) [inline readme.md:63]: please make search-collector monospaced: `search-collector`

**Review by dockerymick — COMMENTED** [review]
Completed the review, please let me know if you need another review thanks!

File: readme.md (hunk 2 above)
Comment (mqhackett) [reply readme.md]: 👍

File: readme.md (hunk 3 above)
Comment (mqhackett) [reply readme.md]: I added punctuation. 
I changed the command to set origin to `git remote add upstream git@github.com:open-cluster-management/search-collector.git`

File: readme.md (hunk 5 above)
Comment (mqhackett) [reply readme.md]: 👍

File: readme.md [hunk 6]
```diff
@@ -49,4 +49,19 @@ RUNTIME_MODE    | no       | production             | Running mode (development
 
 ### Other Configuration Options
 - Environment variables can also be set in ./config.json for development. If both provide a value for a specific property, the environment variable will override the file. You can define your own config json file and pass it to the application with `-c <config_file>`
-- The application can take any flags for [glog](https://github.com/golang/glog), it will pass them straight into glog. The glog flag `--logtostderr` is set to true by default.
\ No newline at end of file
+- The application can take any flags for [glog](https://github.com/golang/glog), it will pass them straight into glog. The glog flag `--logtostderr` is set to true by default.
```
[the remark this answers is not in the document]
Comment (mqhackett) [reply readme.md:52]: Update document title capitalization
changed `search collector` to `search-collector` (added dash)
updated data model section to include link
Updated Usage and configuration capitaliztion

I left the prerequisites section as is. The tips are listed as steps that sometimes are required to fetch the dependencies, step one of the steps.

**Comment by mqhackett** [comment]
@dockerymick @jlpadilla 

I made edits based on comments Mikela left. Thanks again for the review!

File: readme.md [hunk 7]
```diff
@@ -48,5 +49,16 @@ REPORT_RATE_MS  | no       | 5000    // 5 seconds   | Interval(ms) to queue chan
...
+
+### Contribution
+When you make contributions to this project, fork the project then merge your changes by creating a pull request from your personal fork to the main project:
+
+1. Fork the `search-collector` repository.
+2. Clone your fork.
+3. Set upstream origin with the following command: `git remote add upstream git@github.com:open-cluster-management/search-collector.git`
+4. Make your edits. Test locally with `make test` before creating a pull request to ensure smooth merging :smile:
+5. Push your new commits to your personal fork with the following command: `git push origin`
+6. Create a pull request from your personal fork again the upstream `search-collector` main branch
```
Comment (dockerymick) [inline readme.md:64]: @mqhackett should it be from your personal fork **against** the upstream ...?

**Review by dockerymick — CHANGES_REQUESTED** [review]

**Pull request merged**